### DISCUSSION TITLE
Solving the Backdoor challenge in 1 transaction

### DISCUSSION THREAD

**Discussion by Koowah** [tinchoabbate/damn-vulnerable-defi#27] [Help / Q&A / Troubleshooting] [answered]
Hi all,

I managed to get through the challenge with two transactions (deployment + call) using the callback. Can't seem to find a way to achieve it in one transaction however. I thought about doing everything in the constructor but it doesn't seem feasable here.
Anyone managed to do it ?

Also, where is the `getStorageAt` function below coming from ? It's in the `WalletRegistry.sol` file. I can't trace back to its definition.
![cazz](https://user-images.githubusercontent.com/81920095/214583538-486d3832-911e-43fb-8d6e-16d3278a8a97.PNG)

Thanks,
Cheers 😄

**Comment by MiniGlome** [comment, marked as answer]
It's located in `@gnosis.pm/safe-contracts/contracts/common/StorageAccessible.sol`

**Reply by Koowah** [reply]
Thank you @MiniGlome

**Comment by MiniGlome** [comment]
> I managed to get through the challenge with two transactions (deployment + call) using the callback. Can't seem to find a way to achieve it in one transaction however. I thought about doing everything in the constructor but it doesn't seem feasable here. Anyone managed to do it ?

I managed to complete the challenge in one transaction by deploying the backdoored module from the constructor.

**Comment by coderbradlee** [comment]
https://github.com/coderbradlee/damn-vulnerable-defi/blob/dev/contracts/backdoor/testWalletRegistry.sol#L61 look at this if you need hint

**Comment by web3cook** [comment]
The problem with single transaction is that the safefactory calls safe setup as delegate call which in turn calls module as a delegate call. Thus the approve functions lets the access of tokens in safefactory instead of safe of the user itself
